examples/crm/src/dataProvider.ts

```diff
@@ -11,20 +16,45 @@ const TASK_MARKED_AS_UNDONE = 'TASK_MARKED_AS_UNDONE';
 const TASK_DONE_NOT_CHANGED = 'TASK_DONE_NOT_CHANGED';
 let taskUpdateType = TASK_DONE_NOT_CHANGED;
 
+const beforeContactUpsert = async (
+    params: CreateParams<any> | UpdateParams<any>,
+    dataProvider: DataProvider
+) => {
+    const { data } = params;
+    const avatarUrl = await getAvatarUrl(data);
+    data.avatar = avatarUrl || null;
```

Comment:
Youy should never modify a param passed to a function. Instead, create a new object and clone the old data in it.